packages/dumb/src/components/trends/cbg/cbg-slice/cbg-slice-animated.tsx

```diff
@@ -0,0 +1,158 @@
+ */
+
+import cx from 'classnames'
+import React, { FunctionComponent } from 'react'
+import { spring, TransitionMotion } from '@serprex/react-motion'
+
+import { springConfig } from '../../../../models/constants/animation.constants'
+
+import styles from './cbg-slice-animated.css'
+import { CbgSliceTransitionMotionInterpolate } from '../../../../models/animation.model'
+import CbgSliceSegment from './cbg-slice-segment'
+import { TrendsDisplayFlags } from '../../../../models/trends-display-flags.model'
+import { getRangeSegments } from './cbg-slice-animated.util'
+
+interface CbgSliceAnimatedProps {
+  datum: {
+    firstQuartile?: number
+    id: string
+    max?: number
+    median?: number
+    min?: number
+    msX: number
+    ninetiethQuantile?: number
+    tenthQuantile?: number
+    thirdQuartile?: number
+  }
+  displayFlags: TrendsDisplayFlags
+  showingCbgDateTraces: boolean
+  sliceWidth: number
+  tooltipLeftThreshold: number
+  topMargin: number
+  xScale: Function
+  yScale: Function
+}
+
+const DEFAULT_SEGMENT_Y = 16
+
+export const CbgSliceAnimated: FunctionComponent<CbgSliceAnimatedProps> = (props) => {
+  const {
+    datum,
+    displayFlags,
+    showingCbgDateTraces,
+    sliceWidth,
+    tooltipLeftThreshold,
+    topMargin,
+    xScale,
+    yScale
+  } = props
+
+  const strokeWidth = sliceWidth / 8
+  const binLeftX = xScale(datum.msX) - sliceWidth / 2 + strokeWidth / 2
+  const width = sliceWidth - strokeWidth
+
+  const rangeSegments = getRangeSegments(displayFlags)
+
+  const defaultStyles = rangeSegments.map(segment => ({
+    key: segment.key,
+    style: {
+      [segment.y]: DEFAULT_SEGMENT_Y,
+      [segment.height]: 0,
+      opacity: 0
+    }
+  }))
+
+  const transitionMotionStyle = rangeSegments.map(segment => ({
+    key: segment.key,
+    style: {
+      [segment.y]: spring(yScale(datum[segment.y]), springConfig),
+      [segment.height]: spring(
+        yScale(datum[segment.heightKeys[0]]) - yScale(datum[segment.heightKeys[1]]),
+        springConfig
+      ),
+      opacity: spring(1.0, springConfig)
+    }
+  }))
+
+  const yPositions = {
+    firstQuartile: yScale(datum.firstQuartile),
+    max: yScale(datum.max),
+    median: yScale(datum.median),
+    min: yScale(datum.min),
+    ninetiethQuantile: yScale(datum.ninetiethQuantile),
+    tenthQuantile: yScale(datum.tenthQuantile),
+    thirdQuartile: yScale(datum.thirdQuartile),
+    topMargin
+  }
+
+  return (
+    <TransitionMotion
+      defaultStyles={defaultStyles}
+      styles={transitionMotionStyle}
+    >
+      {(interpolatedStyles: CbgSliceTransitionMotionInterpolate[]) => {
+        if (interpolatedStyles.length === 0) {
+          return null
+        }
+        return (
+          <g id={`cbgSlice-${datum.id}`}>
+            {interpolatedStyles.map(interpolated => {
+              const key = interpolated.key
+              const segment = rangeSegments.find(segment => segment.key === key)
```

Comment:
If you only need to know whether the key matches one of the segments (and not retrieve the whole segment), you could do:
```ts
const hasMatchingSegment = rangeSegments.some((segment: xxx) => segment.key === key)
if (!hasMatchingSegment) {
  return null
}
```

It does not change the code a lot but I think it's a bit more readable

Reply:
But I am using `segment` :thinking: